assets/js/components/KeyMetrics/utils.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Key Metrics utility functions.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies.
+ */
+import propTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { trackEvent } from '../../util';
+
+export function conversionReportingDetectedEventsTracking(
+	{
+		shouldShowInitialCalloutForTailoredMetrics,
+		shouldShowCalloutForUserPickedMetrics,
+		haveConversionEventsWithDifferentMetrics,
+		userPickedMetrics,
+		haveLostConversionEvents,
+	},
+	viewContext = '',
+	eventName = ''
+) {
+	// Return early if view context or event name is not present.
+	if ( ! viewContext || ! eventName ) {
+		return;
+	}
```

Comment:
Since they are required, and when we remove optional value, we probably won't need this since it component will throw error if they are not included. Especially `eventName`, which is a static string, and when mandatory it is always going to have a value.

Reply:
@zutigrm this has been addressed in my recent commits, thank you.